### PR TITLE
Fix the way environment is handled for worker_tool processes.

### PR DESCRIPTION
**Pull request by mikekap**
Previously the worker process was started with environment variables from whatever first rule happened to start it.

This changes the behavior to:
 - allow specifying environment variables via `env` in worker_tool
 - remove step-specific environment variables from the worker launch environment.
 - moves `$CWD` to project root
 - moves `$TMP` to `buck-out/bin/<worker_tool target>_worker`

This way the worker is started the same way every time. It also avoids including the
(potentially huge) environment from the genrule in an actual call to exec.

In terms of code organization, I followed the existing pattern, but I think perhaps
it might make sense to move `WorkerProcess` startup into `WorkerTool` itself so we
don't have to pass so many arguments around.

### CODE REVIEW

**Comment by facebook-github-bot-6** [comment]
By analyzing the blame information on this pull request, we identified @sam-swarr, @ryu2 and @andrewjcg to be potential reviewers.

**Comment by facebook-github-bot-2** [comment]
@mikekap updated the pull request.

**Comment by mikekap** [comment]
ping

File: src/com/facebook/buck/shell/WorkerShellStep.java [hunk 1]
```diff
@@ -43,22 +42,16 @@
...
-  private Path tmpPath;
-  private Path workingDir;
   private Optional<WorkerJobParams> cmdParams;
   private Optional<WorkerJobParams> bashParams;
   private Optional<WorkerJobParams> cmdExeParams;
 
   public WorkerShellStep(
       ProjectFilesystem filesystem,
-      Path tmpPath,
-      Path workingDir,
```
Comment (Coneko) [inline src/com/facebook/buck/shell/WorkerShellStep.java]: You can't remove this, it's not always the root of the filesystem, for `genrule` it's the sandbox directory.

File: src/com/facebook/buck/shell/WorkerTool.java [hunk 2]
```diff
@@ -50,12 +59,12 @@ public String getArgs() {
...
+  }
+
   @Override
   public ImmutableSortedSet<BuildRule> getRuntimeDeps() {
-    return ImmutableSortedSet.<BuildRule>naturalOrder()
-        .add(exe)
-        .addAll(exe.getExecutableCommand().getDeps(getResolver()))
-        .addAll(depsFromStartupArgs)
-        .build();
+    return getDeps();
```
Comment (Coneko) [inline src/com/facebook/buck/shell/WorkerTool.java]: Why this change?

**Comment by Coneko** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-6** [comment]
@mikekap updated the pull request.

**Comment by Coneko** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-2** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/131324783940933/int_phab) to review.
